master/internal/api_workspace.go

```diff
@@ -449,6 +449,65 @@ func (a *apiServer) PostWorkspace(
 	}, nil
 }
 
+func (a *apiServer) DeleteWorkspaceNamespaceBindings(ctx context.Context,
+	req *apiv1.DeleteWorkspaceNamespaceBindingsRequest,
+) (*apiv1.DeleteWorkspaceNamespaceBindingsResponse, error) {
+	currUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, c := range req.ClusterNames {
+		err = a.validateRequestClusterName(c)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	err = a.deleteWorkspaceNamespaceBinding(ctx, req.ClusterNames, req.WorkspaceId, currUser)
+	if err != nil {
+		return nil, err
+	}
+	return &apiv1.DeleteWorkspaceNamespaceBindingsResponse{}, nil
+}
+
+func (a *apiServer) deleteWorkspaceNamespaceBinding(ctx context.Context,
+	clusterNames []string, workspaceID int32, curUser *model.User,
+) error {
+	w, err := a.GetWorkspaceByID(ctx, workspaceID, *curUser, true)
+	if err != nil {
+		return err
+	}
+
+	if err := workspace.AuthZProvider.Get().
+		CanSetWorkspaceNamespaceBindings(ctx, *curUser, w); err != nil {
+		return status.Error(codes.PermissionDenied, err.Error())
+	}
+	if len(clusterNames) > 0 {
+		err = db.Bun().RunInTx(ctx, nil, func(ctx context.Context, tx bun.Tx) error {
+			deletedBindings, err := workspace.DeleteWorkspaceNamespaceBindings(ctx, int(workspaceID), clusterNames, &tx)
+			if err != nil {
+				return err
+			}
+			for _, v := range deletedBindings {
+				err = a.m.rm.RemoveEmptyNamespace(v.Namespace, v.ClusterName)
```

Comment:
It threw me a bit that it switched from `Delete` in the API to `Remove` in the Resource Manger. Is the RM using `Remove` because the namespace will still exist (but RM "forgets" about it) so it's not a true delete?

Reply:
Yes exactly. e are just removing the namespace from our interfaces because it will be no longer used by any workspaces, it might still exist in the k8 cluster